### DISCUSSION TITLE
given a node, how to obtain its neighbors?

### DISCUSSION THREAD

**Discussion by RenqinCai** [pyg-team/pytorch_geometric#2707] [Q&A]
I have problems obtaining the neighbors of a specific node. 

I think one way is to convert the graph into networkx data structure and then use the API in networkx to get the neighbors. I feel this is not very efficient. Are there any methods to get the neighbors of a node efficiently? Thanks.

**Comment by rusty1s** [comment]
IMO, the current best way to do this is to convert your `edge_index` into a `SparseTensor` representation and select/slice the nodes that you care about:
```python
from torch_sparse import SparseTensor

adj = SparseTensor.from_edge_index(edge_index)
_, col, _ = adj[0].coo()  # Will return the neighbors of node `0`
row, col, _ = adj[:10].coo()  # Will return the neighbors of node `0, ..., 9`
```

**Reply by RenqinCai** [reply]
Thanks so much for your quick response. Sure. I will give it a try. Thanks.
